### DISCUSSION TITLE
Installing and using "seminar"

### DISCUSSION THREAD

**Discussion by piiskop** [quarto-dev/quarto-cli#4322] [Q&A]
i want to use the seminar plugin with Quarto presentations. So i created a subfolder called seminar inside the folder where i have the presentation QMD-file. i downloaded plugin.js into that folder. i created the file plugin.yml with the following content:

```
name: RevealSeminar
script: [plugin.js]

```

In the presentation file's metadata block, i added:

```
revealjs-plugins:
  - seminar

```

And i clicked the button for rendering. A message box appeared telling me:

> Seminar server not specified!

The manual of this plugin does not state how to create and run my own server so that i can specify it. Please describe how to do that. The presentation file is open but i have no idea how to remote control it from my smartphone. i see no additional buttons and there is no separate file for speaker notes as in multiplex. Please feed some instructions into me.

**Comment by cderv** [comment]
> The manual of this plugin does not state how to create and run my own server so that i can specify it. Please describe how to do that

You should see that with the plugin developer and its community. This is a tool for RevealJS and even if Quarto support revealjs and its plugin, we don't know all of the them and how they works. 

For a quick search : 

* Here information about seminar requirement: https://github.com/rajgoel/reveal.js-plugins/blob/master/seminar/README.md#requirements with links to example 

* Configurations are described in the following paragraph : https://github.com/rajgoel/reveal.js-plugins/blob/master/seminar/README.md#configuration


When you understand how that works, our doc shows you how you should configure (https://quarto.org/docs/presentations/revealjs/advanced.html#reveal-plugins) with links to examples - e.g multiplex code for the built-in plugin (https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/revealjs/plugins/multiplex/plugin.yml) 

You'll find many more example and doc in the Revealjs plugin extension doc https://quarto.org/docs/extensions/listing-revealjs.html

With all that (found using a quick search on web and quarto doc) you should probably make it work. 

Hope it helps

**Reply by piiskop** [reply]
So i went into the requirements section, followed the link to the server folder, cloned it and now the server is running.

So i went into the configuration section, and figured out that the necessary parameters are _server_, _room_, _hash_, and _autoJoin_.

So i went into the plugins section and followed the instructions by creating the subfolder _seminar_ into the subfolder of my presentation, cloned all the plugin files into that created folder, and put the parameters with values into _plugin.yml_:

```
name: RevealSeminar
script: [plugin.js]
config:
  seminar:
    server: 'http://manual-for-rstudio.tennis24.ee:4433/'
    room: 'Some room name'
    hash: '$2a$05$kTSzbyI478NMZV8uN1NrV.LoLukWSL7Qwd6w6PPnIU3ZMXjWHsi4a'
    autoJoin: true 

```

Now, rendering the presentation is successful however i have no idea how to use it. The display itself now has only working menu elements and even the speaker view cannot connect. If _seminar_ works like _multiplex_ then there should be another file for the speaker but there is not. Instead, there is that room. What room? How can i access the presentation as a host, how can i access it as a non-host? i am only interested in getting it to work with _quarto_ as i am managing my presentations with _RStudio Server_.

**Reply by m00sbrugger** [reply]
Piiskop, did you manage to get it to work? I too am using quarto and found this plugin very interesting, but my lack of skills in coding prevents med from going any futher.

**Reply by piiskop** [reply]
No.

**Reply by adam-coates** [reply]
After quite a while of tinkering around with this I managed to get the `seminar` plugin running with quarto. 
However, only after using a post processing script. 

I provide an example [here](https://github.com/adam-coates/seminar-quarto) for those that might find it useful. 

Certain modifications are needed for the menu (a custom tab to enter credentials as an `"admin"`)  and a postprocess `python` script to basically search some strings and replace them with some functions and syntax needed to make it work.
